app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -483,6 +483,47 @@ export default class ComposerService extends Service {
     return uploadIcon(this.currentUser.staff, this.siteSettings);
   }
 
+  @discourseComputed(
+    "model.action",
+    "isWhispering",
+    "model.privateMessage",
+    "model.post.username"
+  )
+  ariaLabel(modelAction, isWhispering, privateMessage, postUsername) {
+    const getLabel = {
+      createSharedDraft: () => I18n.t("composer.create_shared_draft"),
+      editSharedDraft: () => I18n.t("composer.edit_shared_draft"),
+      createTopic: () => I18n.t("composer.composer_actions.create_topic.label"),
+      privateMessage: () => I18n.t("user.new_private_message"),
+      edit: () => I18n.t("composer.composer_actions.edit"),
+      reply: () => {
+        if (isWhispering) {
+          return `${I18n.t("composer.create_whisper")} ${this.site.get(
+            "whispers_allowed_groups_names"
+          )}`;
+        }
+
+        if (privateMessage) {
+          return I18n.t("composer.create_pm");
+        }
+
+        if (postUsername) {
+          return I18n.t("composer.composer_actions.reply_to_post.label", {
+            postUsername,
+          });
+        } else {
+          return I18n.t("composer.composer_actions.reply_to_topic.label");
+        }
+      },
+    };
+
+    if (getLabel[modelAction]) {
+      return getLabel[modelAction]();
+    }
+
+    return I18n.t("keyboard_shortcuts_help.composing.title");
```

Comment:
Small one but you can just use a switch case statement here as there is no need for us to be creating a new object and functions.